src/components/HOC/withScore.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+const withScore = (WrappedComponent, charactersQuantity) => {
+    class WithScore extends React.Component {
+        constructor(props) {
+            super(props)
+            this.state = {
+                score: 0,
+                currentTry: 1,
+                scoreDisplay: false,
+                scoreMessage: '',
+                scoreStyle: 'font--small--red',
+            }
+        }
+
+        toggleScoreDisplay = () => {
+            this.setState({
+                scoreDisplay: !this.state.scoreDisplay
+            });
+        }
+
+        increment = (stateValue) => {
+            this.setState((prevState, ) => {
+                return { [stateValue]: prevState[stateValue] + 1 }
+            });
+        }
+
+        resetGame = () => {
+            this.setState({
+                score: 0,
+                currentTry: 1,
+            });
+        }
+
+        setScoreMessage = (scoreMessage, scoreStyle) => {
+            this.setState({
+                scoreStyle: scoreStyle,
+                scoreMessage: scoreMessage,
+            });
+        }
+
+        render() {
+            return (
+                <WrappedComponent
+                    toggleScoreDisplay={this.toggleScoreDisplay}
+                    scoreDisplay={this.state.scoreDisplay}
```

Comment:
You;re passing all the store;s values to the wrapped component.
It makes it 5 custom properties.
Wouldnt be esier to pass just `state` there?

Reply:
I can do it but then I get linting problems, because I need to specify all props with PropTypes, do you have any suggestions for it?